contracts/abstracts/Minter.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import { Admin } from "./Admin.sol";
+
+/// @title Minter
+/// @dev An abstract contract that implements a basic Minter access control functionality.
+abstract contract Minter is Admin {
+    /// @notice Only minter can access this function
+    error Access_OnlyMinter();
+
+    /// @dev Emitted when the `minter` address is updated.
+    /// @param minter The previous minter address.
+    /// @param newMinter The new minter address.
+    event SetMinter(address indexed minter, address indexed newMinter);
+
+    /// @notice The address of the current minter.
+    address public minter;
+
+    /// @dev Modifier to restrict access to only the minter.
+    modifier onlyMinter() {
+        if (msg.sender != minter) {
+            revert Access_OnlyMinter();
+        }
+        _;
+    }
```

Comment:
instead of storing a minter in storage, can you please use the `MINTER_ROLE` found in base `cre8ors`?

you can follow the pattern found in `src/cre8ing.sol`
<img width="830" alt="Screenshot 2023-08-02 at 1 15 56 AM" src="https://github.com/defientco/crea8ors-subscription-contracts/assets/23249402/c32dd4c6-8145-4220-b443-448193e419f4">